Applications/web_portal/web_app/library/source/pages/account_page/account_page.tsx

```diff
@@ -8,11 +13,257 @@ interface Properties {
 }
 
 interface State {
+  isLoading: boolean;
+  breakpoint?: Breakpoint;
 }
 
+enum Breakpoint {
+  SMALL,
+  MEDIUM,
+  LARGE
+}
+interface UsernameProps {
+  name: string;
+  height: string;
+}
+interface HeaderProps {
+  name: string;
+  roles: Nexus.AccountRoles;
+}
 /** Implements the container used to display account information. */
 export class AccountPage extends React.Component<Properties, State> {
+  public constructor(props: Properties) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      breakpoint: AccountPage.getBreakpoint()
+    };
+    this.onScreenResize = this.onScreenResize.bind(this);
+  }
+
+  public componentWillMount() {
+    this.props.model.load().then(
+      () => {
+        this.setState({
+          isLoading: false
+      });
+    });
+  }
+
+  public componentDidMount() {
+    window.addEventListener('resize', this.onScreenResize);
+  }
+
+  public componentWillUnmount(): void {
+    window.removeEventListener('resize', this.onScreenResize);
+  }
+
   public render(): JSX.Element {
-    return null;
+    const Header = (): JSX.Element => {
```

Comment:
Instead of creating a nested component, use a variable to represent the component you want to display. What's happening now is every time a render takes place, a completely new React Component is being defined on every render.

  const header = (() => {
    ...code...
  })();

  \<Layout\>
    {header}
  \</Layout\>